java/src/main/java/genericpubsub/ManagedSubscribe.java

```diff
@@ -0,0 +1,282 @@
+        serverStream = asyncStub.managedSubscribe(this);
+        ManagedFetchRequest.Builder builder = ManagedFetchRequest.newBuilder().setNumRequested(BATCH_SIZE);
+
+        if (Objects.nonNull(developerName)) {
+            builder.setDeveloperName(developerName);
+        }
+        if (Objects.nonNull(managedSubscriptionId)) {
+            builder.setSubscriptionId(managedSubscriptionId);
+        }
+        serverStream.onNext(builder.build());
+    }
+
+    /**
+     * Helps keep the subscription active by sending FetchRequests at regular intervals.
+     *
+     * @param numOfRequestedEvents
+     */
+    private void fetchMore(int numOfRequestedEvents) {
+        logger.info("Fetching more events: {}", numOfRequestedEvents);
+        ManagedFetchRequest fetchRequest = ManagedFetchRequest
+                .newBuilder()
+                .setNumRequested(numOfRequestedEvents)
+                .build();
+        serverStream.onNext(fetchRequest);
+    }
+
+    /**
+     * Helper function to process the events received.
+     */
+    private void processEvent(ManagedFetchResponse response) throws IOException {
+        if (response.getEventsCount() > 0) {
+            for (ConsumerEvent event : response.getEventsList()) {
+                String schemaId = event.getEvent().getSchemaId();
+                logger.info("processEvent - EventID: {} SchemaId: {}", event.getEvent().getId(), schemaId);
+                Schema writerSchema = getSchema(schemaId);
+                GenericRecord record = deserialize(writerSchema, event.getEvent().getPayload());
+                logger.info("Received event: {}", record.toString());
+            }
+            logger.info("Processed batch of {} event(s)", response.getEventsList().size());
+        }
+
+        // Commit the replay after processing batch of events or commit the latest replay on an empty batch
+        if (!response.hasCommitResponse()) {
+            doCommitReplay(response.getLatestReplayId());
+        }
+    }
+
+    /**
+     * Helper function to commit the latest replay received from the server.
+     */
+    private void doCommitReplay(ByteString commitReplayId) {
+        int numRequested = 1;
+        String newKey =UUID.randomUUID().toString();
+        ManagedFetchRequest.Builder fetchRequestBuilder = ManagedFetchRequest.newBuilder().setNumRequested(numRequested);
+        CommitReplayRequest commitRequest = CommitReplayRequest.newBuilder()
+                .setCommitRequestId(newKey)
+                .setReplayId(commitReplayId)
+                .build();
+        fetchRequestBuilder.setCommitReplayIdRequest(commitRequest);
+
+        logger.info("Sending CommitRequest with numRequested {} , CommitReplayRequest ID: {}", numRequested , newKey);
+        serverStream.onNext(fetchRequestBuilder.build());
+    }
+
+    /**
+     * Helper function to inspect the status of a commitRequest.
+     */
+    private void checkCommitResponse(@NotNull ManagedFetchResponse fetchResponse) {
```

Comment:
Can we remove the NotNull check so we don't need to add another library that we will need to keep updated?

Reply:
This is dependency removed